rust_snuba/src/processors/spans_v2.rs

```diff
@@ -0,0 +1,538 @@
+            segment_id: from
+                .segment_id
+                .map_or(0, |s| u64::from_str_radix(&s, 16).unwrap_or(0)),
+            segment_name: sentry_tags.get("transaction").cloned().unwrap_or_default(),
+            is_segment: from.is_segment,
+            _sort_timestamp: (from.start_timestamp_ms / 1000) as u32,
+            start_timestamp: from.start_timestamp_ms,
+            end_timestamp: from.start_timestamp_ms + from.duration_ms as u64,
+            duration_ms: from.duration_ms,
+            exclusive_time_ms: from.exclusive_time_ms,
+            retention_days: from.retention_days,
+            name: from.description.unwrap_or_default(),
+
+            sampling_weight: 1.,
+            sampling_factor: 1.,
+            sign: 1,
+
+            ..Default::default()
+        };
+
+        {
+            let mut attr_str_buckets = [
+                &mut res.attr_str_0,
+                &mut res.attr_str_1,
+                &mut res.attr_str_2,
+                &mut res.attr_str_3,
+                &mut res.attr_str_4,
+                &mut res.attr_str_5,
+                &mut res.attr_str_6,
+                &mut res.attr_str_7,
+                &mut res.attr_str_8,
+                &mut res.attr_str_9,
+                &mut res.attr_str_10,
+                &mut res.attr_str_11,
+                &mut res.attr_str_12,
+                &mut res.attr_str_13,
+                &mut res.attr_str_14,
+                &mut res.attr_str_15,
+                &mut res.attr_str_16,
+                &mut res.attr_str_17,
+                &mut res.attr_str_18,
+                &mut res.attr_str_19,
+                &mut res.attr_str_20,
+                &mut res.attr_str_21,
+                &mut res.attr_str_22,
+                &mut res.attr_str_23,
+                &mut res.attr_str_24,
+                &mut res.attr_str_25,
+                &mut res.attr_str_26,
+                &mut res.attr_str_27,
+                &mut res.attr_str_28,
+                &mut res.attr_str_29,
+                &mut res.attr_str_30,
+                &mut res.attr_str_31,
+                &mut res.attr_str_32,
+                &mut res.attr_str_33,
+                &mut res.attr_str_34,
+                &mut res.attr_str_35,
+                &mut res.attr_str_36,
+                &mut res.attr_str_37,
+                &mut res.attr_str_38,
+                &mut res.attr_str_39,
+                &mut res.attr_str_40,
+                &mut res.attr_str_41,
+                &mut res.attr_str_42,
+                &mut res.attr_str_43,
+                &mut res.attr_str_44,
+                &mut res.attr_str_45,
+                &mut res.attr_str_46,
+                &mut res.attr_str_47,
+                &mut res.attr_str_48,
+                &mut res.attr_str_49,
+            ];
+
+            sentry_tags.iter().chain(tags.iter()).for_each(|(k, v)| {
+                attr_str_buckets[(half_md5(k.as_bytes()) as usize) % attr_str_buckets.len()]
```

Comment:
Are we sure we can use the same hashing mechanism on the querying side?

Reply:
"half_md5" should be the hash with the best language support, it's in clickhouse too